src/rnnlm/rnnlm-utils.cc

```diff
@@ -20,20 +21,45 @@
 #include <numeric>
 #include "rnnlm/rnnlm-utils.h"
 
+
 namespace kaldi {
 namespace rnnlm {
 
 
 void ReadSparseWordFeatures(std::istream &is,
                             int32 feature_dim,
                             SparseMatrix<BaseFloat> *word_feature_matrix) {
-  // TODO.
-}
-
+  std::vector<std::vector<std::pair<MatrixIndexT, BaseFloat> > > sparse_rows;
+  std::string line;
+  int32 line_number = 0;
+  while (std::getline(is, line)) {
+    std::vector<std::pair<MatrixIndexT, BaseFloat> > row;
+    std::istringstream line_is(line);
+    int32 word_id;
+    line_is >> word_id;
+    line_is >> std::ws;
+    KALDI_ASSERT(word_id == line_number++);
```

Comment:
For things that could be a problem in user input, better to fail with KALDI_ERR and an informative error message, instead of KALDI_ASSERT.